language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -565,6 +567,52 @@ writePackageIdTs dir pkgId =
     T.writeFileUtf8 (dir </> "packageId.ts") $ T.unlines
       ["export default '" <> unPackageId pkgId <> "';"]
 
+writeIndexTs :: FilePath -> [Module] -> IO ()
+writeIndexTs packageSrcDir modules =
+  T.writeFileUtf8 (packageSrcDir </> "index.ts") lines
+  where
+    lines :: T.Text
+    lines = T.unlines $ header <> concatMap entry modules
+
+    header :: [T.Text]
+    header =
+      [ "import {default as __packageId} from \"./packageId\""
+      , "export const packageId = __packageId"
+      ]
+
+    entry :: Module -> [T.Text]
+    entry mod
+      | null $ defDataTypes mod = []
+      | otherwise = importDecl var path <> exportDecl var parts
+      where
+        parts = unModuleName (moduleName mod)
+        var = "__" <> T.intercalate "_" parts
+        path =  "\"./" <> T.intercalate "/" parts <> "\""
+
+    importDecl :: T.Text -> T.Text -> [T.Text]
+    importDecl var path =
+      [ "/* eslint-disable @typescript-eslint/camelcase */"
+      , "import * as " <> var <> " from "  <> path
+      ]
+
+    exportDecl :: T.Text -> [T.Text] -> [T.Text]
+    exportDecl = go 0
+      where
+        spaces i = T.pack $ replicate i ' '
+
+        go :: Int -> T.Text -> [T.Text] -> [T.Text]
+        go indent var [m] = let ws = spaces indent in
+          [ ws <> "/* eslint-disable @typescript-eslint/no-unused-vars */"
+          , ws <> "export import " <> m <> " = " <> var
```

Comment:
I don’t understand this. Why is the variable unused if it’s exported? https://eslint.org/docs/2.0.0/rules/no-unused-vars#exporting-variables claims that this is only required if you use `var` which doesn’t seem to be the case.

Reply:
i don't know. i can tell you though that in practice lint warnings are generated.
```
$ eslint --ext .ts --max-warnings 0 src/

/private/var/folders/kc/bjk2hzwx6bv07jz_s80wjh7w0000gn/T/extra-dir-43326669297/daml2ts/057eed1fd48c238491b8ea06b9b5bf85a5d4c9275dd3f6183e0e6b01730cc2ba/src/index.ts
  9:19  warning  'Down' is defined but never used  @typescript-eslint/no-unused-vars
```